app/views/protocols/import_export/_import_json_protocol_preview_modal.html.erb

```diff
@@ -0,0 +1,338 @@
+              display_last_modified.to_s+"(Protocols.io value)",readonly: true, class:
+              "form-control" %>
+            </div>
+          </div>
+        </div>
+
+        <!-- Preview title -->
+        <div>
+          <h2 style="display: inline;"><%= t("protocols.import_export.import_modal.preview_title") %></h2>
+          <h3 style="display: none;" data-role="title-position"></h3>
+        </div>
+
+        <!-- Preview scroller -->
+
+        <div
+          class="import-protocols-modal-preview-container-json"
+          data-role="preview-container">
+          <hr>
+
+          <% if @json_object["before_start"]&&@json_object["before_start"]!="" %>
+
+          <strong>Before starting protocol information:</strong><br>
+          <%= (@json_object["before_start"]) %><br>
+
+          <% end %>
+          <% if @json_object["warning"]&&@json_object["warning"]!="" %>
+
+          <strong>Protocol warning:</strong><br>
+          <%= (@json_object["warning"]) %><br>
+
+          <% end %>
+          <% if @json_object["guidelines"]&&@json_object["guidelines"]!="" %>
+
+          <strong>Guidelines:</strong><br>
+          <%= (@json_object["guidelines"]) %><br>
+
+          <% end %>
+
+          <% if @json_object["link"]&&@json_object["link"]!="" %>
+
+          <br>
+          <strong>Supplied link:</strong><br>
+          <%= (@json_object["link"]) %>
+
+          <% end %>
+
+          <%  counter=0 %>
+          <%  whitelist_simple=["1","6","17"] %>
+          <%  whitelist_complex=["8","9","15","18","19","20"]%>
+          <% @json_object["steps"].each do |step| %>
+          <div style="display: block;">
+            <hr>
+            <td>
+              <div class="badge-num">
+                <span class="badge-preview bg-primary size-digit-1">
+                  <b data-val="position"><%= (counter+=1) %></b>
+                </span>
+                &nbsp; &nbsp;
+                <span class="step-panel-collapse-link" data-toggle="collapse">
+                  <span class="glyphicon collapse-step-icon glyphicon-collapse-up"></span>
+                  <%# this guid attributes were ommited because they didnt seem relevenat %>
+                  <%#=   Step guid (Protocols.io) :%><%#= step["guid"] %>
+                  <%# if counter>1 #only steps after the first one have previous step guid field
+                  valid%>
+                  <%#=Guid of previous step (Protocols.io) :%><%#= step["previous_guid"] %>
+                  <%# end %>
+                  <% title=nil %>
+                  <% step["components"].each do |key1,value1| #finding section (title of step) %>
+
+                  <% if value1.class==Hash %>
+                  <% key1=value1 %>
+                  <% end %>
+
+                  <%  if(key1["component_type_id"]=="6") %>
+                  <% if(!key1["data"].nil? && key1["data"]!="") %>
+                  <% title ||=key1["data"] %>
```

Comment:
Here you just set the value:
<% title = key1["data"] %>

Reply:
Then you could just make one liner instead of previous two lines:
<% title = key1["data"] if key1["data"].present? %>